pkg/config/config.go

```diff
@@ -13,16 +13,22 @@ const SectionKey = "server"
 
 type ServerConfig struct {
 	HTTPPort             int                   `json:"httpPort" pflag:",On which http port to serve admin"`
-	GrpcPort             int                   `json:"grpcPort" pflag:",On which grpc port to serve admin"`
-	GrpcServerReflection bool                  `json:"grpcServerReflection" pflag:",Enable GRPC Server Reflection"`
+	GrpcPort             int                   `json:"grpcPort" pflag:",deprecated"`
+	GrpcServerReflection bool                  `json:"grpcServerReflection" pflag:",deprecated"`
 	KubeConfig           string                `json:"kube-config" pflag:",Path to kubernetes client config file, default is empty, useful for incluster config."`
 	Master               string                `json:"master" pflag:",The address of the Kubernetes API server."`
 	Security             ServerSecurityOptions `json:"security"`
-
+	GrpcConfig           GrpcConfig            `json:"grpc"`
 	// Deprecated: please use auth.AppAuth.ThirdPartyConfig instead.
 	DeprecatedThirdPartyConfig authConfig.ThirdPartyConfigOptions `json:"thirdPartyConfig" pflag:",Deprecated please use auth.appAuth.thirdPartyConfig instead."`
 }
 
+type GrpcConfig struct {
+	Port                    int  `json:"port" pflag:",On which grpc port to serve admin"`
+	ServerReflection        bool `json:"serverReflection" pflag:",Enable GRPC Server Reflection"`
+	MaxGrpcMessageSizeBytes int  `json:"maxMessageSizeBytes" pflag:",The max size in bytes for incoming gRPC messages"`
```

Comment:
Nit, drop the redundant "Grpc"
```suggestion
	MaxMessageSizeBytes int  `json:"maxMessageSizeBytes" pflag:",The max size in bytes for incoming gRPC messages"`
```

Reply:
d'oh thanks! this is what i get for prefix matching